internal/proto/local/controller/storage/host/plugin/store/v1/host.proto

```diff
@@ -42,9 +42,15 @@ message HostCatalog {
   // @inject_tag: `gorm:"default:null"`
   uint32 version = 8;
 
+  // secrets_hmac is a sha256-hmac of the unencrypted secrets that is returned
+  // from the API for read.  It is recalculated everytime the raw secrets are
+  // updated.
+  // @inject_tag: `gorm:"default:null"`
+  string secrets_hmac = 9;
```

Comment:
This should probably be bytes.  We can't use use the value for anything except to return it.

Reply:
It goes in as a base64'd string. Is it still useful to put it in as bytes? I can, just not sure of the difference.

Reply:
The token_hmac for credential_vault_token is stored as bytes (see [this](https://github.com/hashicorp/boundary/blob/main/internal/proto/local/controller/storage/credential/vault/store/v1/vault.proto#L79) and [this](https://github.com/hashicorp/boundary/blob/main/internal/db/schema/migrations/oss/postgres/10/04_vault_credential.up.sql#L148) ) as well as most other hmacs (except maybe an oidc hmac that might has slipped through the cracks)

Reply:
Yes, we can only use the value as a base64 encoded string, so why store it in mem or the db in any other format?

Reply:
Why are we base64 encoding it? The hmac is binary data not a string. That fact that we are encoding binary data into a string doesn't change the fact that the data itself is binary data.

Reply:
while true, the data only has one format for the entire domain, which is a string not bytes.

Reply:
Any string is binary data :-D

Reply:
After discussing with @jimlambrt, we recommend this be switched to bytes.